src/libAtomVM/otp_net.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * This file is part of AtomVM.
+ *
+ * Copyright 2023 by Fred Dushin <fred@dushin.net>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+ */
+
+#include <context.h>
+#include <defaultatoms.h>
+#include <globalcontext.h>
+#include <interop.h>
+#include <nifs.h>
+#include <otp_net.h>
+#include <port.h>
+#include <term.h>
+
+#include <errno.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+// #define ENABLE_TRACE
+#include <trace.h>
+
+#define FAMILY_ATOM globalcontext_make_atom(global, ATOM_STR("\x6", "family"))
+#define PROTOCOL_ATOM globalcontext_make_atom(global, ATOM_STR("\x8", "protocol"))
+#define TYPE_ATOM globalcontext_make_atom(global, ATOM_STR("\x4", "type"))
+#define INET_ATOM globalcontext_make_atom(global, ATOM_STR("\x4", "inet"))
+#define ADDR_ATOM globalcontext_make_atom(global, ATOM_STR("\x4", "addr"))
```

Comment:
This is already a static const in otp_socket.c
I'm concerned we are creating copies of the same constants.
Did we completely give up on defaultatoms?

Reply:
I am not fond of default atoms as a dumping ground, as on some platforms these atoms don't have any use and just occupy memory.

I agree there is a lot of redundancy between this and opt-socket, but I also wanted this module to be self-contained, partly to avoid difficult merges, but also because it is independent of any socket code (though likely may not have a use-case outside of the context of a socket)

Reply:
Following @bettio 's advice, I have in-lined the calls (no macro)